File: Prgrammers_LV1/옹알이.cpp
#include <string>
#include <vector>
#include<iostream>
#include<algorithm>
using namespace std;

int check_babble(string babble)
{
	vector<string> words = {"aya", "ye", "woo", "ma"};
	string tmp = "";
	string last_word = "";
	for (auto ch : babble)
	{
		tmp += ch;
		if (tmp == last_word)
			return 0;
		if (find(words.begin(), words.end(),tmp) != words.end())
		{
			last_word = tmp;
			tmp = "";
		}
	}
	if (tmp == "")
		return 1;
	return 0;
}

int solution(vector<string> babbling) {
    int answer = 0;
	for (auto babble : babbling)
		answer += check_babble(babble);
    return answer;
}

int main()
{
	vector<string> babbling = {"aya", "yee", "u", "maa"};
	cout << solution(babbling);
}

File: Python/옹알이.py
def check_babble(babble):
    words = ["aya", "ye","woo","ma"]
    tmp = ""
    last = ""
    for ch in babble:
        tmp += ch
        if tmp in words:
            if last == tmp:
                return False
            last = tmp
            tmp = ""
    if len(tmp) == 0:
        return True
    return False
    
def solution(babbling):
    answer = 0
    for babble in babbling:
        answer += check_babble(babble)
    return answer